fem/tmop_tools.cpp

```diff
@@ -638,6 +747,47 @@ void TMOPNewtonSolver::ProcessNewState(const Vector &x) const
          }
       }
    }
+
+   if (update_surf_fit_coeff)
+   {
```

Comment:
You should add 2 sentences about the general idea of the adaptive weight, or add more comments to the important lines below.